src/BlueberryUpgrades.ts

```diff
@@ -0,0 +1,444 @@
+    } else {
+      return 'Contact Platonic or Khafra if you see this (should never occur!)'
+    }
+  }
+
+  public get bonus() {
+    return this.rewards(this.level)
+  }
+}
+
+export const blueberryUpgradeData: Record<keyof Player['blueberryUpgrades'], IBlueberryData> = {
+  ambrosiaTutorial: {
+    maxLevel: 10,
+    costPerLevel: 1,
+    blueberryCost: 0,
+    costFormula: (level: number, baseCost: number): number => {
+      return baseCost * (Math.pow(level + 1, 2) - Math.pow(level, 2))
+    },
+    rewards: (n: number) => {
+      const cubeAmount = 1 + 0.05 * n
+      const quarkAmount = 1 + 0.01 * n
+      return {
+        quarks: quarkAmount,
+        cubes: cubeAmount,
+        desc: String(i18next.t('ambrosia.data.ambrosiaTutorial.effect', { cubeAmount: format(100 * (cubeAmount - 1), 0, true), quarkAmount: format(100 * (quarkAmount - 1), 0, true) }))
+      }
+    }
+  },
+  ambrosiaQuarks1: {
+    maxLevel: 100,
+    costPerLevel: 1,
+    blueberryCost: 0,
+    costFormula: (level: number, baseCost: number): number => {
+      return baseCost * (Math.pow(level + 1, 3) - Math.pow(level, 3))
+    },
+    rewards: (n: number) => {
+      const quarkAmount = 1 + 0.01 * n
+      return {
+        quarks: quarkAmount,
+        desc: String(i18next.t('ambrosia.data.ambrosiaQuarks1.effect', { amount: format(100 * (quarkAmount - 1), 0, true) }))
+      }
+    },
+    prerequisites: {
+      'ambrosiaTutorial': 10
+    }
+  },
+  ambrosiaCubes1: {
+    maxLevel: 100,
+    costPerLevel: 1,
+    blueberryCost: 0,
+    costFormula: (level: number, baseCost: number): number => {
+      return baseCost * (Math.pow(level + 1, 3) - Math.pow(level, 3))
+    },
+    rewards: (n: number) => {
+      const cubeAmount = (1 + 0.05 * n) * Math.pow(1.1, Math.floor(n / 10))
+      return {
+        cubes: cubeAmount,
+        desc: String(i18next.t('ambrosia.data.ambrosiaCubes1.effect', { amount: format(100 * (cubeAmount - 1), 2, true) }))
+      }
+    },
+    prerequisites: {
+      'ambrosiaTutorial': 10
+    }
+  },
+  ambrosiaLuck1: {
+    maxLevel: 100,
+    costPerLevel: 1,
+    blueberryCost: 0,
+    costFormula: (level: number, baseCost: number): number => {
+      return baseCost * (Math.pow(level + 1, 3) - Math.pow(level, 3))
+    },
+    rewards: (n: number) => {
+      const val = 2 * n + 5 * Math.floor(n/10)
+      return {
+        ambrosiaLuck: val,
+        desc: String(i18next.t('ambrosia.data.ambrosiaLuck1.effect', { amount: format(val) }))
+      }
+    },
+    prerequisites: {
+      'ambrosiaTutorial': 10
+    },
+    cacheUpdates: [() => player.caches.ambrosiaLuck.updateVal('BlueberryUpgrade1')]
+  },
+  ambrosiaQuarkCube1: {
+    maxLevel: 25,
+    costPerLevel: 250,
+    blueberryCost: 1,
+    costFormula: (level: number, baseCost: number): number => {
+      return baseCost * (Math.pow(level + 1, 3) - Math.pow(level, 3))
+    },
+    rewards: (n: number) => {
+      const baseVal = 0.0005 * n
+      function val(base: number) {
```

Comment:
I don't see a case where val is needed (same goes for all below upgrades too), it's being called twice with the same value every time. It could be simplified to

```js
const val = 1 + baseVal * Math.floor(Math.pow(Math.log10(Number(player.worlds)+1) +1, 2))
```

(etc.) for the ones below